static/app/views/explore/hooks/useMetricOptions.tsx

```diff
@@ -4,51 +4,59 @@ import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilte
 import type {PageFilters} from 'sentry/types/core';
 import {DiscoverDatasets} from 'sentry/utils/discover/types';
 import {useApiQuery, type ApiQueryKey} from 'sentry/utils/queryClient';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
 import useOrganization from 'sentry/utils/useOrganization';
 import usePageFilters from 'sentry/utils/usePageFilters';
-
-interface EventsMetricResult {
-  data: Array<{
-    ['metric.name']: string;
-    ['metric.type']: string;
-  }>;
-  meta?: {
-    fields?: Record<string, string>;
-  };
-}
+import {
+  TraceMetricKnownFieldKey,
+  type TraceMetricEventsResult,
+} from 'sentry/views/explore/metrics/types';
 
 interface UseMetricOptionsProps {
   datetime?: PageFilters['datetime'];
   enabled?: boolean;
+  orgSlug?: string;
   projectIds?: PageFilters['projects'];
+  search?: string;
 }
 
 function metricOptionsQueryKey({
   orgSlug,
   projectIds,
   datetime,
-}: {
-  orgSlug: string;
-  datetime?: PageFilters['datetime'];
-  projectIds?: number[];
-}): ApiQueryKey {
+  search,
+}: UseMetricOptionsProps = {}): ApiQueryKey {
+  const searchValue = new MutableSearch('');
+  if (search) {
+    searchValue.addStringContainsFilter(
+      `${TraceMetricKnownFieldKey.METRIC_NAME}:${search}`
+    );
+  }
   const query: Record<string, string | string[] | number[]> = {
     dataset: DiscoverDatasets.TRACEMETRICS,
-    field: ['metric.name', 'metric.type', 'count(metric.name)'],
+    field: [
+      TraceMetricKnownFieldKey.METRIC_NAME,
+      TraceMetricKnownFieldKey.METRIC_TYPE,
+      TraceMetricKnownFieldKey.METRIC_UNIT,
+      `count(${TraceMetricKnownFieldKey.METRIC_NAME})`,
+    ],
+    query: searchValue.formatString(),
     referrer: 'api.explore.metric-options',
-    orderby: 'metric.name',
   };
 
   if (projectIds?.length) {
     query.project = projectIds.map(String);
   }
 
-  if (datetime) {
+  if (search && datetime) {
+    // If searching we use the full filters in order to not miss the result.
     Object.entries(normalizeDateTimeParams(datetime)).forEach(([key, value]) => {
       if (value !== undefined) {
         query[key] = value as string | string[];
       }
     });
+  } else {
+    query.statsPeriod = '1h'; // Default to a much smaller time window if not searching.
```

Comment:
I'm inclined to say we should raise this to say 24 hours to capture metrics that are emitted hourly